airdrop/application/services/user_registration_services.py

```diff
@@ -1,77 +1,85 @@
+            airdrop_window = AirdropWindowRepository().get_airdrop_window_by_id(airdrop_window_id)
 
             if airdrop_window is None:
                 raise Exception("Invalid Airdrop window id")
 
-            is_eligible_user, rewards_awards = self.check_user_eligibility(
-                user_address=address, airdrop_id=airdrop_id, airdrop_window_id=airdrop_window_id)
+            user_eligible_for_given_window = UserRepository(). \
+                is_user_eligible_for_given_window(address, airdrop_id, airdrop_window_id)
+
+            rewards_awarded = AirdropRepository().fetch_total_rewards_amount(airdrop_id, address)
 
-            is_already_registered, registration_id = self.is_elgible_registered_user(
-                airdrop_window_id, address)
+            user_registered, user_registration = UserRepository(). \
+                get_user_registration_details(address, airdrop_window_id)
 
             is_airdrop_window_claimed = False
-            airdrop_claim_status = self.is_airdrop_window_claimed(
-                airdrop_window_id, address)
+            is_claimable = False
+            airdrop_claim_status = AirdropWindowRepository().is_airdrop_window_claimed(airdrop_window_id, address)
 
             if airdrop_claim_status == AirdropClaimStatus.SUCCESS.value:
                 is_airdrop_window_claimed = True
-            is_claimable = False
+            else:
+                if rewards_awarded > 0:
+                    is_claimable = True
             # if the user has not claimed yet and there are rewards pending to be claimed , then let the user claim
             # rewards awarded will have some value ONLY when the claim window opens and the user has unclaimed rewards
             # a claim in progress ~ PENDING will also be considered as claimed ( we don't want the user to end up losing
             # gas in trying to claim again)
-            if is_airdrop_window_claimed is False and rewards_awards > 0:
-                is_claimable = True
-            reject_reason = None
-            if not is_eligible_user:
-                reject_reason = UserRepository().get_reject_reason(airdrop_window_id, address)
-
-            response = AirdropWindowEligibility(airdrop_id, airdrop_window_id, address, is_eligible_user,
-                                                is_already_registered, is_airdrop_window_claimed, airdrop_claim_status,
-                                                reject_reason, rewards_awards, registration_id, is_claimable).to_dict()
 
+            if user_registered:
+                registration_id = user_registration.registration_id
+                reject_reason = user_registration.reject_reason
+                registration_details = {
+                    "registration_id": user_registration.registration_id,
+                    "reject_reason": user_registration.reject_reason,
+                    "other_details": user_registration.signed_data.get("message", {}),
```

Comment:
can you also correct the signed_data to signed_params ? as discussed on the call ysterday ?